src.ts/multisig.ts

```diff
@@ -0,0 +1,306 @@
+
+            let multisig: MultiSigWalletProperties = checkFormat({
+                owner: checkString,
+                threshold: checkNumber,
+                signers: checkAny,
+            }, properties);
+
+            let transaction = signer.provider.getTransactionRequest("multisig", "auth-createMultiSigAccount", {
+                from: multisig.owner,
+                threshold: multisig.threshold,
+                signers: multisig.signers
+            });
+            transaction.fee = signer.provider.getTransactionFee(undefined, undefined, { tx: transaction });
+            
+            return signer.sendTransaction(transaction, overrides).then((response) => {
+                
+                let groupAddress = deriveAddress(signerAddress, signer.getNonce());
+
+                if (overrides && overrides.sendOnly) {
+                    return response;
+                }
+                let confirmations = (overrides && overrides.confirmations) ? Number(overrides.confirmations) : null;
+
+                return signer.provider.waitForTransaction(response.hash, confirmations).then((receipt) => {
+                    if (1 == receipt.status) {
+                        return new MultiSigWallet(groupAddress, signer);
+                    }
+                    throw signer.provider.checkTransactionReceipt(receipt, errors.CALL_EXCEPTION, "create multisig wallet failed", {
+                        method: "auth-createMultiSigAccount",
+                        receipt
+                    });
+                });
+            });
+        });
+    }
+
+    /**
+     * Update multiSig wallet
+     * @param groupAddress group address
+     * @param threshold number of threshold
+     * @param signers signers
+     */
+    static update(properties: UpdateMultiSigWalletProperties, signer: Signer, overrides?: any): Promise<TransactionResponse | TransactionReceipt> {
+        if (!Signer.isSigner(signer)) {
+            errors.throwError('update multisig wallet transaction require signer', errors.MISSING_ARGUMENT, { arg: 'signer' });
+        }
+        if (properties && 'string' === typeof (properties)) {
+            properties = JSON.parse(properties);
+        }
+
+        checkProperties(properties, {
+            from: true,
+            groupAddress: true,
+            threshold: true,
+            signers: true,
+        }, true);
+
+        return resolveProperties({ address: signer.getAddress() }).then(({ address }) => {
+            if (!address) {
+                errors.throwError('update multisig wallet require signer address', errors.MISSING_ARGUMENT, { required: 'signer' });
+            }
+            properties.from = address; // Set signer address as owner
+
+            let multisig: UpdateMultiSigWalletProperties = checkFormat({
+                from: checkString,
+                groupAddress: checkString,
+                threshold: checkNumber,
+                signers: checkString,
```

Comment:
The `signers` should be a list of address. Should we use the `arrayOf(checkAddress)`?

Reply:
fxied